vm/io.cpp

```diff
@@ -13,6 +13,39 @@ The Factor library provides platform-specific code for Unix and Windows
 with many more capabilities so these words are not usually used in
 normal operation. */
 
+int raw_fread(void* ptr, size_t size, size_t nitems, FILE* stream) {
+  size_t items_read = 0;
+
+  do {
+    size_t ret = fread((void*)((int*)ptr + items_read * size), size,
+                       nitems - items_read, stream);
+    if (ret == 0) {
```

Comment:
The man pages say:

| If an error occurs, or the end-of-file is reached, the return value is a short object count (or zero).

So perhaps you need to check ``if (ret <= expected)``.